platform/viewer/src/lib/DICOMSR/srUtils.js

```diff
@@ -0,0 +1,71 @@
+import OHIF from '@ohif/core';
+
+const { studyMetadataManager } = OHIF.utils;
+
+const supportedSopClassUIDs = [
+  '1.2.840.10008.5.1.4.1.1.88.22',
+  '1.2.840.10008.5.1.4.1.1.11.1',
+];
+
+const getAllDisplaySets = () => {
+  const allStudies = studyMetadataManager.all();
+  let allDisplaySets = [];
+
+  allStudies.forEach(study => {
+    if (study.getDisplaySets) {
+      allDisplaySets = allDisplaySets.concat(study.getDisplaySets());
+    }
+  });
+
+  return allDisplaySets;
+};
+
+const getInstanceMetadata = (displaySets, sopInstanceUid) => {
+  let instance;
+
+  // find the correct instance
+  displaySets.some(displaySet => {
```

Comment:
is `some` faster than `find`?

Reply:
I've update the code and remove part of the logic and make it more simple.